src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceHyperResultSet.java

```diff
@@ -160,7 +158,7 @@ private void getNextChunk() throws SQLException {
         log.trace("Fetching page with number {} for resultset {}", ++currentPageNum, this);
         try {
             List<Object> rows = arrowUtil.getRowsFromStreamResponse();
-            if (rows != null && rows.size() > 0) {
```

Comment:
Don't we need `rows.size()>0` check?

Reply:
I don't see why we would need it. An empty table with 0 rows is a perfectly valid table.